src/lesson28/Server.java

```diff
@@ -0,0 +1,71 @@
+package lesson28;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Scanner;
+
+public class Server {
+    private static final int MAX_CLIENTS = 3;
+    private static List<ClientHandler> clients = Collections.synchronizedList(new ArrayList<>());
+
```

Comment:
If list is already synchronized, then there is no need in additional monitor for section above. Possibly, to pass it as synchronized list instead of abstraction will be better. Also should be better to use composition instead of agregation for this list